test/device/neighbours_test.cpp

```diff
@@ -0,0 +1,53 @@
+#include <gtest/gtest.h>
+
+#include <algorithm>  // Include for std::shuffle
+#include <random>     // Include for random number generation
+
+#include "utils.hpp"
+
+namespace test {
+
+class Neighbours : public testing::Test {
+public:
+    void TearDown() override {};
+    void SetUp() override {};
+};
+
+TEST_F(Neighbours, NeighboursAreCalculatedCorrectly) {
+    auto source = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
+
+    size_t NUMBER_OF_NEIGHBOURS = 3;
+    auto target_neighbours = createRoutingModules(NUMBER_OF_NEIGHBOURS);
+    size_t NUMBER_OF_DESTINATIONS = 5;
+    auto dests = createRoutingModules(NUMBER_OF_DESTINATIONS);
+
+    auto link1_neighbour1 =
+        std::make_shared<TestLink>(TestLink(source, target_neighbours[0]));
+    auto link2_neighbour1 =
+        std::make_shared<TestLink>(TestLink(source, target_neighbours[0]));
+    auto link3_neighbour1 =
+        std::make_shared<TestLink>(TestLink(source, target_neighbours[0]));
+    auto link1_neighbour2 =
+        std::make_shared<TestLink>(TestLink(source, target_neighbours[1]));
+    auto link1_neighbour3 =
+        std::make_shared<TestLink>(TestLink(source, target_neighbours[2]));
+
+    EXPECT_TRUE(source->get_neighbours().empty());
+
+    source->update_routing_table(dests[0], link2_neighbour1);
+    source->update_routing_table(dests[1], link1_neighbour3);
+    source->update_routing_table(dests[2], link1_neighbour2);
+    source->update_routing_table(dests[3], link3_neighbour1);
+    source->update_routing_table(dests[4], link1_neighbour1);
```

Comment:
It is very easy now to get confused with such links names, they are too similar. I think you may create function `create_random_links(std::shared_ptr<sim::RoutingModule> source, std::vector<sim::RoutingModule> neighbours, size_t links_count)`, that generates `link_count` random links from `source` to `neighbors`. Also it allows to replace 5 lines with `source->update_routing_table(...)` with one easily for understanding `for`:
```suggestion
    auto links = create_rangom_links(source, target_neighbours, NUMBER_OF_DESTINATIONS);
    

    EXPECT_TRUE(source->get_neighbours().empty());

    for (size_t i = 0; i < NUMBER_OF_DESTINATIONS; i++) {
        source->update_routing_table(dests[i], links[i]);
    }
```

Reply:
I was trying to do something like this, but at some moment I made it too complicated and dropped this idea
I will try again